gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -547,8 +547,28 @@ case class InsertColumnarToColumnarTransitions(session: SparkSession) extends Ru
       })
   }
 
+  private def replaceWithColumnarToColumnar(plan: SparkPlan): SparkPlan = {
+    plan match {
+      case p: RowToColumnarExecBase if p.child.isInstanceOf[ColumnarToRowExec] =>
+        val replacedChild = replaceWithColumnarToColumnar(
+          p.child.asInstanceOf[ColumnarToRowExec].child)
+        BackendsApiManager.getSparkPlanExecApiInstance.genColumnarToColumnarExec(replacedChild)
```

Comment:
The case condition says it matches `vanilla-columnar->row->gluten-columnar` but it calls a general interface to do `columnar->columnar`. Do we need to specialize the interface name to make sure we exactly request for `vanilla-columnar->gluten-columnar` transition here?

Reply:
Thanks for the explanation.

If it's for ColumnarToRow + RowToVeloxColumnar,

Would it be better to rename the method `genColumnarToColumnarExec` with `genVanillaColumnarToGlutenColumnarExec` or something?

Reply:
So it's clearer that we are not doing `Gluten` -> `Vanilla`  c2c transition here.